subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildBuildPathAssignmentIntegrationTest.groovy

```diff
@@ -182,22 +213,34 @@ class CompositeBuildBuildPathAssignmentIntegrationTest extends AbstractComposite
         fixture.progress(BuildIdentifiedProgressDetails).findAll { it.details.buildPath }*.details*.buildPath
     }
 
-    private void configureBuildConfigurationOutput(List<BuildTestFile> builds) {
+    private void configureNestedBuild(List<BuildTestFile> builds) {
         builds.each {
-            it.buildFile << """
-                    subprojects {
-                        // The Java plugin forces the configuration of the included builds
-                        apply plugin: 'java'
+            it.settingsFile << """
+                gradle.projectsEvaluated {
+                    // only print from the root build of the build tree to avoid duplication
+                    if (gradle.owner.identityPath.path == ':') {
+                        def registry = gradle.services.get(${BuildStateRegistry.name})
+                        registry.visitBuilds { b ->
+                            def buildId = b.buildIdentifier
+                            println "Build name=" + buildId.name
```

Comment:
❓ Apart from the name, we'd also need some assertions about `isCurrentBuild()`, given that we don't have any integration tests for it. Is this available from the `BuildStateRegistry` or is it coming from somewhere else?